Automattic/wp-calypso
# Removes the [emails/show-new-comparison-component] feature flag

This is a follow-up to #65088, and is the next stage in utilising the new mailbox forms. This set of changes ensure that the stacked email comparison view is now used for the domain upsell, and finally releases references to the older mailbox components.

The email checkout process is also modified to make room for the upsell process where the associated domain is still in the cart alongside the email subscription.

#### Proposed Changes

* Feature flag `emails/show-new-comparison-component` is dropped
* `<EmailProvidersComparison/>` is dropped as well. To keep the PR light and easy to review, components and symbols that were obsoleted as a result of retiring the component will be removed in a follow-up.
* A new selector to provide the appropriate product based on the email provider and billing interval. This is necessary because a subscription does not exist at this stage.
* Some new [design improvements for the upsell view](pbLl1t-1a5-p2#comment-964) are also incorporated.

#### Testing Instructions



| Upsell before | Upsell after |
|--------|-------|
| <img width="750" alt="Screenshot 2022-07-03 at 8 45 37 AM" src="https://user-images.githubusercontent.com/277661/177030339-370b3acd-f8e8-4ace-a6b9-d3cc761555a8.png">      | <img width="750" alt="Screenshot 2022-07-03 at 8 45 51 AM" src="https://user-images.githubusercontent.com/277661/177030345-29410d66-d1db-44cc-95dd-1e929b7ee794.png">     |


* Select a new domain for purchase
* Confirm that you are presented with a domain upsell featuring the `<EmailProvidersStackedComparison/>` component. The `Components` tab of the [Redux dev tools](https://github.com/reduxjs/redux-devtools) can be used to confirm this.
* Confirm the new design for the domain upsell in the above screenshot
* Confirm that design for new email subscription purchases is retained ( i.e. non domain-upsell view )
* Confirm that the [ **Back** ] and [ **Skip** ] navigation buttons work as desired
* Confirm for each of the billing intervals i.e. ( monthly, annual ), that the accurate product billing interval is displayed in the shopping cart before checkout.
* Confirm that you can purchase either a Professional Email or a Google Workspace subscription successfully, and that the mailboxes are provisioned correctly.
* Confirm that additional mailboxes can be added from the Email plan page or otherwise. ( This regression test ensures that existing correct behaviour isn't lost ).
